packages/devtools_app/lib/src/ui/flutter/vm_flag_widgets.dart

```diff
@@ -42,6 +47,23 @@ class ProfileGranularityDropdownState
         if (safeValue != flag.valueAsString) {
           _onProfileGranularityChanged(safeValue, context);
         }
+
+        final bannerMessages = BannerMessages.of(context);
+        if (safeValue == highProfilePeriod) {
+          bannerMessages.push(
+              HighProfileGranularityMessage(widget.screenType).build(context));
+        } else {
+          final currentMessages =
```

Comment:
this business logic should be in the BannerMessages controller and matching should not need to be done by calling toString on keys.

Reply:
done